pkg/metrics/eventmetrics/collector.go

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+package eventmetrics
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/tetragon/pkg/api/processapi"
+	"github.com/cilium/tetragon/pkg/option"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var metricsLabels = map[int]string{
+	processapi.MetricExec:       "exec",
+	processapi.MetricClone:      "clone",
+	processapi.MetricExit:       "exit",
+	processapi.MetricData:       "data",
+	processapi.MetricCgroup:     "cgroup",
+	processapi.MetricLoader:     "loader",
+	processapi.MetricTracepoint: "tracepoint",
+	processapi.MetricKprobe:     "kprobe",
+	processapi.MetricUprobe:     "uprobe",
+}
+
+// bpfCollector implements prometheus.Collector. It collects metrics directly from BPF maps.
+type bpfCollector struct{}
+
+func NewBPFCollector() prometheus.Collector {
+	return &bpfCollector{}
+}
+
+func (c *bpfCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- MissedEvents.Desc()
+}
+
+func (c *bpfCollector) Collect(ch chan<- prometheus.Metric) {
+	mapHandle, err := ebpf.LoadPinnedMap(filepath.Join(option.Config.MapDir, "tg_stats_map"), nil)
+	if err != nil {
+		return
+	}
+	defer mapHandle.Close()
+
+	var zero uint32
+	var allCpuValue []processapi.KernelStats
+	if err := mapHandle.Lookup(zero, &allCpuValue); err != nil {
+		return
+	}
+
+	sum := processapi.KernelStats{}
+	for _, val := range allCpuValue {
+		for i := 0; i < processapi.MetricMaxValues; i++ {
+			sum.Sent[i] += val.Sent[i]
+			sum.SentFailed[i] += val.SentFailed[i]
+		}
+	}
+
+	for i := 0; i < processapi.MetricMaxValues; i++ {
+		if sum.Sent[i] > 0 {
+			ch <- MissedEvents.MustMetric(float64(sum.Sent[i]), fmt.Sprintf("%s_sent", metricsLabels[i]))
```

Comment:
If I understand correctly, "sent" events are not "missed", so it's confusing that they're reported by `MissedEvents`. I would rather define another metric, `SentEvents` (`tetragon_sent_events_total`) for those. Then we can label metrics by the event type, without `_sent` and `_failed` suffixes.